3d2ce73 break unit tests

Unit tests break with 3d2ce73. Also, when running `make test`, the .yaml files are updated and generated differently from the checked in files.

I can fix this by reverting the commit

~~~
[akaris@linux ingress-node-firewall ((0ec1511...))]$ git checkout downstream/master
Warning: you are leaving 1 commit behind, not connected to
any of your branches:

  0ec1511 Revert "fix go get controller-gen issue"

If you want to keep it by creating a new branch, this may be a good time
to do so with:

 git branch <new-branch-name> 0ec1511

HEAD is now at c0a24f2 Merge pull request #104 from msherif1234/xdp_list_intfs
[akaris@linux ingress-node-firewall ((c0a24f2...))]$ git checkout -b revert-3d2ce73
Switched to a new branch 'revert-3d2ce73'
[akaris@linux ingress-node-firewall (revert-3d2ce73)]$ make test
==== Generating DaemonSet manifest
hack/generate-daemon-manifest.sh
/home/akaris/development/go/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/akaris/development/ingress-node-firewall/api/v1alpha1/ingressnodefirewallnodestate_types.go:31:35: map values must be a named type, not *ast.ArrayType
Error: not all generators ran successfully
run `controller-gen rbac:roleName=manager-role crd webhook paths=./... output:crd:artifacts:config=config/crd/bases -w` to see all available markers, or `controller-gen rbac:roleName=manager-role crd webhook paths=./... output:crd:artifacts:config=config/crd/bases -h` for usage
make: *** [Makefile:99: manifests] Error 1
[akaris@linux ingress-node-firewall (revert-3d2ce73)]$ git diff
diff --git a/config/crd/bases/ingressnodefirewall.openshift.io_ingressnodefirewallconfigs.yaml b/config/crd/bases/ingressnodefirewall.openshift.io_ingressnodefirewallconfigs.yaml
index bc0c639..9dec8d8 100644
--- a/config/crd/bases/ingressnodefirewall.openshift.io_ingressnodefirewallconfigs.yaml
+++ b/config/crd/bases/ingressnodefirewall.openshift.io_ingressnodefirewallconfigs.yaml
@@ -1,9 +1,10 @@
+
 ---
-apiVersion: apiextensions.k8s.io/v1
+apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.9.0
+    controller-gen.kubebuilder.io/version: v0.3.0
   creationTimestamp: null
   name: ingressnodefirewallconfigs.ingressnodefirewall.openshift.io
 spec:
@@ -14,83 +15,90 @@ spec:
     plural: ingressnodefirewallconfigs
     singular: ingressnodefirewallconfig
   scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      description: IngressNodeFirewallConfig is the Schema for the ingressnodefirewallconfigs
[akaris@linux ingress-node-firewall (revert-3d2ce73)]$ 
~~~

And after revert:
~~~
[akaris@linux ingress-node-firewall (revert-3d2ce73)]$ git stash
Saved working directory and index state WIP on revert-3d2ce73: c0a24f2 Merge pull request #104 from msherif1234/xdp_list_intfs
[akaris@linux ingress-node-firewall (revert-3d2ce73)]$ git revert 3d2ce73
[revert-3d2ce73 4a7e6cb] Revert "fix go get controller-gen issue"
 1 file changed, 3 insertions(+), 14 deletions(-)
[akaris@linux ingress-node-firewall (revert-3d2ce73)]$ git status
On branch revert-3d2ce73
nothing to commit, working tree clean
[akaris@linux ingress-node-firewall (revert-3d2ce73)]$ make test
==== Generating DaemonSet manifest
hack/generate-daemon-manifest.sh
/home/akaris/development/ingress-node-firewall/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/akaris/development/ingress-node-firewall/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
KUBEBUILDER_ASSETS="/home/akaris/.local/share/kubebuilder-envtest/k8s/1.24.1-linux-amd64" go test ./... -coverprofile cover.out
?   	github.com/openshift/ingress-node-firewall	[no test files]
?   	github.com/openshift/ingress-node-firewall/api/v1alpha1	[no test files]
?   	github.com/openshift/ingress-node-firewall/cmd	[no test files]
^Cmake: *** [Makefile:115: test] Error 1

[akaris@linux ingress-node-firewall (revert-3d2ce73)]$ git diff
[akaris@linux ingress-node-firewall (revert-3d2ce73)]$ 
~~~

And when I do:
~~~
rm -f ~/development/go/bin/controller-gen
~~~

I get:
~~~
[akaris@linux ingress-node-firewall ((c0a24f2...))]$ make test
/usr/bin/which: no controller-gen in (/usr/local/go/bin:/usr/local/go/bin:/home/akaris/.local/bin:/home/akaris/bin:/usr/share/Modules/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/akaris/.composer/vendor/bin:/home/akaris/development/go/bin:/home/akaris/development/go/bin)
go: creating new go.mod: module tmp
go: added github.com/fatih/color v1.12.0
go: added github.com/go-logr/logr v1.2.0
go: added github.com/gobuffalo/flect v0.2.5
go: added github.com/gogo/protobuf v1.3.2
go: added github.com/google/gofuzz v1.1.0
go: added github.com/inconshreveable/mousetrap v1.0.0
go: added github.com/json-iterator/go v1.1.12
go: added github.com/mattn/go-colorable v0.1.8
go: added github.com/mattn/go-isatty v0.0.12
go: added github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: added github.com/modern-go/reflect2 v1.0.2
go: added github.com/spf13/cobra v1.4.0
go: added github.com/spf13/pflag v1.0.5
go: added golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3
go: added golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd
go: added golang.org/x/sys v0.0.0-20220209214540-3681064d5158
go: added golang.org/x/text v0.3.7
go: added golang.org/x/tools v0.1.10-0.20220218145154-897bd77cd717
go: added golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: added gopkg.in/inf.v0 v0.9.1
go: added gopkg.in/yaml.v2 v2.4.0
go: added gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: added k8s.io/api v0.24.0
go: added k8s.io/apiextensions-apiserver v0.24.0
go: added k8s.io/apimachinery v0.24.0
go: added k8s.io/klog/v2 v2.60.1
go: added k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
go: added sigs.k8s.io/controller-tools v0.9.0
go: added sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2
go: added sigs.k8s.io/structured-merge-diff/v4 v4.2.1
go: added sigs.k8s.io/yaml v1.3.0
==== Generating DaemonSet manifest
hack/generate-daemon-manifest.sh
/home/akaris/development/go/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
bash: line 1: /home/akaris/development/go/bin/controller-gen: No such file or directory
make: *** [Makefile:99: manifests] Error 127
[akaris@linux ingress-node-firewall ((c0a24f2...))]$ 
~~~

--- a/Makefile
+++ b/Makefile
@@ -240,20 +240,9 @@ $(KUSTOMIZE): $(LOCALBIN)
 	curl -s $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(LOCALBIN)
 
 .PHONY: controller-gen
-controller-gen: ## Download controller-gen locally if necessary.
-ifeq (, $(shell which controller-gen))
-	@{ \
-	set -e ;\
-	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
-	cd $$CONTROLLER_GEN_TMP_DIR ;\
-	go mod init tmp ;\
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION) ;\
-	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
-	}
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessary.
+$(CONTROLLER_GEN): $(LOCALBIN)
+	GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
 
 ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
 

